main.tf

```
@@ -1 +1,445 @@
+# ----------------------------------------------------------------------------------------------------------------------
+# REQUIRE A SPECIFIC TERRAFORM VERSION OR HIGHER
+# This module has been updated with 0.12 syntax, which means it is no longer compatible with any versions below 0.12.
+# ----------------------------------------------------------------------------------------------------------------------
+terraform {
+  required_version = ">= 0.12"
+}
 
+###
+### LOCALS
+###
+
+locals {
+  vpc_id                 = element(data.aws_subnet.lb.*.vpc_id, 0)
+  archive_file_name      = "salt.zip"
+  configs_file_name      = "configs.zip"
+  appscript_file_name    = "appscript.sh"
+  config_dir_path        = "/etc/vault/configs"
+  logs_path              = "/var/log/vault"
+  default_enabled_repos  = ["epel"]
+  default_inbound_cdirs  = ["10.0.0.0/16", "10.0.0.0/8"]
+  s3_bucket_name         = join("-", [var.name, random_string.this.result])
+  appscript_url          = join("/", [module.s3_bucket.id, local.appscript_file_name])
+  archive_dir_path       = join("/", [path.module, ".files"])
+  appscript_dir_path     = join("/", [path.module, "scripts"])
+  role_name              = join("-", [upper(var.name), "INSTANCE", data.aws_caller_identity.current.account_id])
+  ssm_root_path          = join("/", ["vault", var.environment, data.aws_caller_identity.current.account_id, var.name])
+  s3_salt_vault_content  = join("/", [module.s3_bucket.id, local.archive_file_name])
+  s3_vault_configuration = var.vault_configs_path == null ? "" : join("/", [module.s3_bucket.id, local.configs_file_name])
+  dynamodb_table         = var.dynamodb_table == null ? join("", aws_dynamodb_table.this.*.id) : var.dynamodb_table
+  kms_key_id             = var.kms_key_id == null ? join("", aws_kms_key.this.*.id) : var.kms_key_id
+  certificate_arn        = var.certificate_arn == null ? join("", aws_acm_certificate.this.*.id) : var.certificate_arn
+  vault_url              = var.vault_url == null ? join(".", [var.name, var.domain_name]) : var.vault_url
+
+  # Logs files to be streamed to CloudWatch Logs
+  logs = [
+    join("/", [local.logs_path, "state.vault.log"]),
+    join("/", [local.logs_path, "state.vault.initialize.log"]),
+    join("/", [local.logs_path, "state.vault.sync.log"])
+  ]
+
+  tags = merge(var.tags,
```

might as well add name here, and let the user override (specify var.tags last)

```suggestion
  tags = merge(
    {
      Name = var.name
      Environment = var.environment
    },
    var.tags
  )
```